pkg/dao/repository_configs.go

```diff
@@ -372,7 +372,35 @@ func (r repositoryConfigDaoImpl) filteredDbForList(OrgID string, filteredDB *gor
 
 	if filterData.Status != "" {
 		statuses := strings.Split(filterData.Status, ",")
-		filteredDB = filteredDB.Where("status IN ?", statuses)
+
+		filteredDB = filteredDB.
+			Joins("LEFT JOIN tasks ON repository_configurations.last_snapshot_task_uuid = tasks.id").
+			Preload("Repository").
+			Preload("LastSnapshotTask")
+
+		for _, status := range statuses {
+			status = strings.ToLower(status)
+			if status == "valid" {
+				filteredDB = filteredDB.Where("repositories.last_introspection_status = 'Valid' AND tasks.type = 'snapshot' AND tasks.status = 'completed'")
+			}
+			if status == "pending" {
+				filteredDB = filteredDB.Where(
+					"repositories.last_introspection_status = 'Pending' AND (repository_configurations.last_snapshot_task_uuid IS NULL OR (tasks.type = 'snapshot' AND (tasks.status = 'running' OR tasks.status = 'completed')))").
+					Or("repositories.last_introspection_status = 'Valid' AND repository_configurations.last_snapshot_uuid IS NULL AND tasks.type = 'snapshot' AND tasks.status = 'running'")
+			}
+			if status == "unavailable" {
+				filteredDB = filteredDB.Where(
+					"repositories.last_introspection_status = 'Unavailable' AND repository_configurations.last_snapshot_uuid IS NOT NULL AND tasks.type = 'snapshot' AND (tasks.status = 'failed' OR tasks.status = 'completed')").
+					Or("repositories.last_introspection_status = 'Invalid' AND repository_configurations.last_snapshot_uuid IS NOT NULL AND tasks.type = 'snapshot' AND tasks.status = 'failed'").
+					Or("repositories.last_introspection_status = 'Valid' AND repository_configurations.last_snapshot_uuid IS NOT NULL AND tasks.type = 'snapshot' AND (tasks.status = 'running' OR tasks.status = 'failed')")
+			}
+			if status == "invalid" {
+				filteredDB = filteredDB.Where(
+					"repositories.last_introspection_status = 'Invalid' AND tasks.type = 'snapshot' AND tasks.status = 'completed'").
+					Or("repositories.last_introspection_status = 'Unavailable' AND repository_configurations.last_snapshot_uuid IS NULL AND tasks.type = 'snapshot' AND tasks.status = 'failed'").
+					Or("repositories.last_introspection_status = 'Valid' AND repository_configurations.last_snapshot_uuid IS NULL AND tasks.type = 'snapshot' AND tasks.status = 'failed'")
+			}
+		}
```

Comment:
sorry for the delay!   I think what you have is real close, the part that you were missing, is that with multiple statuses, you'd want the equivalent of something like:
```
filteredDB.Where(  db.Where(STMT1).Or(db.Where(STMT2)).Or(db.Where(STMT3))   )
```

I played around with it to make sure it worked (This pulls the statement generation out to a method as well):

```
	if filterData.Status != "" {
		statuses := strings.Split(filterData.Status, ",")

		filteredDB = filteredDB.
			Joins("LEFT JOIN tasks ON repository_configurations.last_snapshot_task_uuid = tasks.id").
			Preload("Repository").
			Preload("LastSnapshotTask")

		var filterChain *gorm.DB
		for _, status := range statuses {
			if filterChain == nil {
				// first where statement, so generate just it
				filterChain = getStatusFilter(status, r.db)
			} else {
				// after the first, OR it with the previous statements
				filterChain = filterChain.Or(getStatusFilter(status, filteredDB))
			}
		}
		filteredDB = filteredDB.Where(filterChain)
	}
	return filteredDB
}

func getStatusFilter(status string, filteredDB *gorm.DB) *gorm.DB {
	status = strings.ToLower(status)
	if status == "valid" {
		filteredDB = filteredDB.Where("(repositories.last_introspection_status = 'Valid' AND tasks.type = 'snapshot' AND tasks.status = 'completed')")
	}
	if status == "pending" {
		filteredDB = filteredDB.Where(
			"repositories.last_introspection_status = 'Pending' AND (repository_configurations.last_snapshot_task_uuid IS NULL OR (tasks.type = 'snapshot' AND (tasks.status = 'running' OR tasks.status = 'completed')))").
			Or("repositories.last_introspection_status = 'Valid' AND repository_configurations.last_snapshot_uuid IS NULL AND tasks.type = 'snapshot' AND tasks.status = 'running'")
	}
	if status == "unavailable" {
		filteredDB = filteredDB.Where(
			"repositories.last_introspection_status = 'Unavailable' AND repository_configurations.last_snapshot_uuid IS NOT NULL AND tasks.type = 'snapshot' AND (tasks.status = 'failed' OR tasks.status = 'completed')").
			Or("repositories.last_introspection_status = 'Invalid' AND repository_configurations.last_snapshot_uuid IS NOT NULL AND tasks.type = 'snapshot' AND tasks.status = 'failed'").
			Or("repositories.last_introspection_status = 'Valid' AND repository_configurations.last_snapshot_uuid IS NOT NULL AND tasks.type = 'snapshot' AND (tasks.status = 'running' OR tasks.status = 'failed')")
	}
	if status == "invalid" {
		filteredDB = filteredDB.Where(
			"repositories.last_introspection_status = 'Invalid' AND tasks.type = 'snapshot' AND tasks.status = 'completed'").
			Or("repositories.last_introspection_status = 'Unavailable' AND repository_configurations.last_snapshot_uuid IS NULL AND tasks.type = 'snapshot' AND tasks.status = 'failed'").
			Or("repositories.last_introspection_status = 'Valid' AND repository_configurations.last_snapshot_uuid IS NULL AND tasks.type = 'snapshot' AND tasks.status = 'failed'")
	}
	return filteredDB
}
```

Reply:
with this, the tests run, but the results fail because i presume the test needs updating